File: cafenamesa-ordering/src/entities/Item.ts
import { randomUUID } from "crypto";

class Item {
    id: String;
    name: String;
    description: String;
    price: Number;
    available: Boolean;
    menu_id: String;
    created_at: String;
    updated_at: String;

    constructor() {
        if(!this.id){
            this.id = randomUUID();
            this.available = true;
        }
    }
}

export { Item };

File: restaurant/frontend/app/models/Item.py
from dataclasses import dataclass
from typing import Dict


@dataclass
class Item:
    id: str
    name: str
    description: str
    price: float
    available: bool
    menu_id: str
    created_at: str
    update_at: str

    def to_dict(self) -> Dict:
        return {
            'id': self.id,
            'name': self.name,
            'description': self.description,
            'price': self.price,
            'available': self.available,
            'menu_id': self.menu_id,
            'created_at': self.created_at,
            'update_at': self.update_at,
        }
